cmake/HPX_AddExecutable.cmake

```diff
@@ -121,7 +121,9 @@ macro(add_hpx_executable name)
       get_filename_component(extension ${source} EXT)
       if(${extension} STREQUAL ".cu")
         message(${extension})
-        SET_SOURCE_FILES_PROPERTIES(${source} PROPERTIES LANGUAGE CXX)
+        SET_SOURCE_FILES_PROPERTIES(${source} PROPERTIES
+          LANGUAGE CXX
+          COMPILE_FLAGS ${HPX_CUDA_CLANG_FLAGS})
       endif()
```

Comment:
Here is a trick to manage files that has .cu extension because with clang we are not using `cuda_add_executable()` but the native `add_executable()`. `SET_SOURCE_FILES_PROPERTIES()` was already used in master